cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -15,22 +12,44 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   import configuration.{includeStack, includeThread}
 
+  def verify(recordingFile: RecordingFile): Boolean = {
+
+    val allEventsTypes = recordingFile.readEventTypes();
+
+    val eventTypes = allEventsTypes.stream().map(eventType => eventType.getLabel).collect(Collectors.toSet[String])
+
+    if(!eventTypes.contains("Allocation in new TLAB")){
+      throw new JfrEventNotFoundException("'Allocation in new TLAB' Event not found")
+    }
+    if(!eventTypes.contains("Allocation outside TLAB")){
+      throw new JfrEventNotFoundException("'Allocation outside TLAB' Event not found")
+    }
+    if(!eventTypes.contains("Method Profiling Sample")){
+      throw new JfrEventNotFoundException("'Method Profiling Sample' Event not found")
+    }
+    return true;
+  }
+
   def parse(): Unit = {
 
+
+
     println(s"*** File $file")
     try {
-      val recording = FlightRecordingLoader.loadFile(file)
-      for (t <- recording.getEventTypes.asScala) {
-        println(t.getName)
-      }
-      val view = recording.createView
-      view.asScala.find(_.getEventType.getName == "GC TLAB Configuration") match {
-        case Some(e) => handleTlabConfiguration(e)
-        case _ => ???
+      val recording = new RecordingFile(file.toPath);
+
+      try{
+        verify(recording)
```

Comment:
this means that you read the file twice
you are just recordng that you saw an event, which can be done, when you see the event ~ line 50